mail_send_copy/models/ir_mail_server.py

```diff
@@ -14,8 +17,19 @@ class IrMailServer(models.Model):
     def send_email(self, message, *args, **kwargs):
         do_not_send_copy = self.env.context.get("do_not_send_copy", False)
         if not do_not_send_copy:
-            if message["Bcc"]:
-                message["Bcc"] = message["Bcc"].join(COMMASPACE, message["From"])
+            # Get existing Bcc recipients (if any)
+            bcc = message.get("Bcc", "")
+            from_addr = message.get("From", "")
+
+            # Combine existing Bcc with From address
+            if bcc:
+                all_bcc = COMMASPACE.join([bcc, from_addr])
             else:
-                message["Bcc"] = message["From"]
+                all_bcc = from_addr
+
+            # Set the combined Bcc
+            message.replace_header(
+                "Bcc", all_bcc
+            ) if "Bcc" in message else message.add_header("Bcc", all_bcc)
```

Comment:
why don't you use a normal conditional here?

Reply:
just free style :) and sure, I will update it